app/services/chart_builder.rb

```diff
@@ -31,33 +31,61 @@ def self.construct_from_cache(cache_read, benchmark_result_type)
     chart_builder
   end
 
-  def initialize(benchmark_runs, benchmark_result_type)
+  def initialize(benchmark_runs, benchmark_result_type, comparing_runs = [])
     @benchmark_result_type = benchmark_result_type
     @benchmark_runs = benchmark_runs
     @columns = {}
+    @comparing_runs = comparing_runs
   end
 
   def build_columns
-    @benchmark_runs.each do |benchmark_run|
-      if block_given?
-        version = yield(benchmark_run)
-        @categories ||= []
-        @categories << version if version != @categories.last
+    if @comparing_runs.present?
+      (@benchmark_runs + @comparing_runs)
+      .sort_by{ |benchmark_run| benchmark_run.initiator.created_at }
```

Comment:
As @tgxworld suggested earlier, you could indent this by a level.

Also, to be honest I didn't test it but there should be a way to make this method a little bit easier to read. What do you think about something like:

~~~ruby
def build_columns
  columns = {}

  if @comparing_runs.present?
    runs = (@benchmark_runs + @comparing_runs)
             .sort_by { |run| run.initiator.created_at }
  else
    runs = @benchmark_runs
  end
    
  runs.each do |run|
    version = nil
    if block_given?
      version = yield(run)
      @categories ||= []
      @categories << version if version != @categories.last
    end

    run.result.each do |key, value|
      if @comparing_runs.present?
        columns["#{key}_#{run.benchmark_type.category}"] ||= []
        columns["#{key}_#{run.benchmark_type.category}"] << [version, value.to_f]
      else
        columns[key] ||= []
        columns[key] << value.to_f
      end
    end
  end

  @columns = columns.map do |name, data|
    { name: name, data: data}
  end

  self
end
~~~